### DISCUSSION TITLE
Provide a `initState` builder helper

### DISCUSSION THREAD

**Discussion by pikax** [histoire-dev/histoire#93] [Ideas]
Since the `initState` should be usually based on the component props, it would be nice to have an out-of-box helper typed function

```ts
import { createInitState } from 'histoire/client'
import  MyComponent from './MyComponent.vue'

const initState = createInitState<typeof MyComponent>({ /* props */ })
```

Once https://github.com/vuejs/core/pull/2693 is merged we can add that type into the `slot:state`

**Comment by Akryum** [comment]
Probably related to this issue: https://github.com/histoire-dev/histoire/issues/31

**Comment by Akryum** [comment]
Automatic prop detection and controls are now implemented: https://github.com/histoire-dev/histoire/issues/31#issuecomment-1141514996
